CRM/Utils/Geocode/Google.php

```diff
@@ -82,7 +82,7 @@ public static function format(&$values, $stateName = FALSE) {
       $add .= ',+';
     }
 
-    if (!empty($values['state_province']) || !empty($values['state_province_id'])) {
+    if (!empty($values['state_province']) || (!empty($values['state_province_id']) && $values['state_province_id'] != 'null')) {
```

Comment:
@jitendrapurohit do you know we have a special function to handle the weirdness that is CiviCRM thinking the four letters "null" actually is a special form of `NULL`?

See [CRM_Utils_System::isEmpty()](https://github.com/civicrm/civicrm-core/blob/5cad71c0ab894d3b64cfef0ded930265d3300695/CRM/Utils/System.php#L684)